src/hal/src/buffer.rs

```diff
@@ -48,66 +48,110 @@ impl Error for ViewError {
 bitflags!(
     /// Buffer usage flags.
     #[cfg_attr(feature = "serialize", derive(Serialize, Deserialize))]
-    pub flags Usage: u16 {
+    pub struct Usage: u16 {
         ///
-        const TRANSFER_SRC  = 0x1,
+        const TRANSFER_SRC  = 0x1;
         ///
-        const TRANSFER_DST = 0x2,
+        const TRANSFER_DST = 0x2;
         ///
-        const UNIFORM = 0x4,
+        const UNIFORM = 0x4;
         ///
-        const STORAGE = 0x8,
+        const STORAGE = 0x8;
         ///
-        const UNIFORM_TEXEL = 0x10,
+        const UNIFORM_TEXEL = 0x10;
         ///
-        const STORAGE_TEXEL = 0x20,
+        const STORAGE_TEXEL = 0x20;
         ///
-        const INDEX = 0x40,
+        const INDEX = 0x40;
         ///
-        const INDIRECT = 0x80,
+        const INDIRECT = 0x80;
         ///
-        const VERTEX = 0x100,
+        const VERTEX = 0x100;
     }
 );
 
+///
+pub const TRANSFER_SRC: Usage   = Usage::TRANSFER_SRC;
```

Comment:
this is great! finally, we have namespaces for bit flags :)

Reply:
Yes, this makes code cleaner. Maybe we can rename some flags properly later.